Sources/Hummingbird/Router/ResponseGenerator.swift

```diff
@@ -81,3 +73,36 @@ extension Optional: HBResponseGenerator where Wrapped: HBResponseGenerator {
         }
     }
 }
+
+public struct HBTypedResponse<Body: HBResponseGenerator>: HBResponseGenerator {
+    public var status: HTTPResponseStatus?
+    public var headers: HTTPHeaders
+    public var body: Body
```

Comment:
Is `body` a correct naming here? It's the entire response, except we potentially override properties

Reply:
How about 
```swift
public struct HBEditedResponse<Generator: HBResponseGenerator>: HBResponseGenerator {
    public var status: HTTPResponseStatus?
    public var headers: HTTPHeaders
    public var responseGenerator: Generator

    public init(
        status: HTTPResponseStatus? = nil,
        headers: HTTPHeaders = .init(),
        response: Generator
    ) {
        self.status = status
        self.headers = headers
        self.responseGenerator = response
    }
```

Reply:
That is a lot better to me